roles/maps/tasks/download_large_file.yml

```diff
@@ -35,33 +35,62 @@
 #
 # TODO - files should be owned by www-data
 
-- name: "Download {{ iiab_map_host_url }}/{{ item }} via .meta4 file"
-  # cd to the temp directory so that the aria2c file, data file, and (perhaps
-  # eventually) torrent file all end up there.
-  shell: |
-    set -euo pipefail
+- block:
+  - name: "Fetching file size for {{ iiab_map_host_url }}/{{ item }} via .meta4 file"
+    shell: |
+      import sys, xmltodict, requests, os
 
-    cd /library/downloads/maps/
+      if os.path.exists("{{ dest_path }}"):
+        sys.exit(0)
 
-    aria2c \
-        --connect-timeout="{{ download_timeout }}" \
-        --log-level=warn \
-        --console-log-level=warn \
-        --summary-interval=0 \
-        --show-console-readout=false \
-        --download-result=hide \
-        --follow-metalink=mem \
-        --max-connection-per-server=4 \
-        --file-allocation=falloc \
-        --enable-http-pipelining=true \
-        --seed-time=0 \
-        "{{ iiab_map_host_url }}/{{ item }}.meta4"
+      size = int(xmltodict.parse(
+        requests.get("{{ iiab_map_host_url }}/{{ item }}.meta4").content
+      )['metalink']['file']['size'])
+      KiB = 1024
+      MiB = KiB * 1024
+      GiB = MiB * 1024
 
-    chmod 644 "{{ item }}"
-    mv "{{ item }}" "{{ dest_path }}"
-  args:
-    executable: /bin/bash
-    creates: "{{ dest_path }}"
+      if size < MiB:
+        size_disp = str(round(size / KiB, 2)) + ' KiB'
+      elif size < GiB:
+        size_disp = str(round(size / MiB, 2)) + ' MiB'
+      else:
+        size_disp = str(round(size / GiB, 2)) + ' GiB'
+      print(' (' + size_disp + ') To check progress, run: "tail {{ working_dir }}/{{ log_file }}"')
+    args:
+      executable: /usr/bin/python3
+    register: download_file_size
+
+  - name: "Download {{ item }}{{ download_file_size.stdout_lines.0 | default(' (done)')}}"
+    # cd to the temp directory so that the aria2c file, data file, and (perhaps
+    # eventually) torrent file all end up there.
+    shell: |
+      set -euo pipefail
+
+      cd {{ working_dir }}
+
+      aria2c \
+          --connect-timeout="{{ download_timeout }}" \
+          --log-level=warn \
+          --console-log-level=warn \
+          --summary-interval=600 \
+          --download-result=hide \
+          --follow-metalink=mem \
+          --max-connection-per-server=4 \
+          --file-allocation=falloc \
+          --enable-http-pipelining=true \
+          --seed-time=0 \
+          "{{ iiab_map_host_url }}/{{ item }}.meta4" \
+          > "{{ log_file }}"
```

Comment:
@chapmanjacobd Normal logging doesn't seem to output the summaries.

Reply:
you're saying that `>` is working but `--log=` is not?

I'm not completely surprised by this... I guess you would need `--log-level=notice` or something instead of `warn`--but it might be much too verbose compared to stdout `summary-interval` so I think you've landed on a good configuration.

Annoying that aria2c doesn't print the first `summary-interval` progress summary sooner though!